test/fixtures/scenarioManager.js

```diff
@@ -0,0 +1,81 @@
+var path = require('path');
+var config = require('./config');
+var elasticsearch = require('elasticsearch');
+
+function ScenarioManager(server) {
+  if (!server) throw new Error('No server defined');
+
+  this.client = new elasticsearch.Client({
+    host: server
+  });
+}
+
+/**
+* Load a testing scenario
+* @param {string} id The scenario id to load
+* @return {Promise} A promise that is resolved when elasticsearch has a response
+*/
+ScenarioManager.prototype.load = function (id) {
+  var scenario = config[id];
+  if (!scenario) throw new Error('No scenario found for ' + id);
+
+  var self = this;
+  return Promise.all(scenario.bulk.map(function bulk(file) {
+    var mapping;
+
+    if (scenario.mapping) {
+      mapping = self.client.indices.create({
+        index: file,
+        body: require(path.join(scenario.base, scenario.mapping))
+      });
+    } else {
+      mapping = Promise.resolve();
+    }
+
+    return mapping.then(function () {
+      self.client.bulk({
+        body: require(path.join(scenario.base, file)),
+      });
+    });
+
+  }));
+};
+
+/**
+* Delete a scenario
+* @param {string} index
+* @return {Promise} A promise that is resolved when elasticsearch has a response
+*/
+ScenarioManager.prototype.unload = function (id) {
+  var scenario = config[id];
+  if (!scenario) throw new Error('Expected index');
+
+  return this.client.indices.delete({
+    index: scenario.bulk
```

Comment:
The object in the config file suggests that the strings used in `mapping` and `bulk` are arbitrary file names. I didn't notice until I got to here that those files must follow the convention `indexName.js` maybe make this explicit somehow?